Sources/Components/SelectorTitleView/SelectorTitleView.swift

```diff
@@ -0,0 +1,126 @@
+    }
+
+    public weak var delegate: SelectorTitleViewDelegate?
+
+    public var arrowDirection: ArrowDirection = .down {
+        didSet {
+            let asset: FinniversImageAsset = arrowDirection == .up ? .arrowUpSmall : .arrowDownSmall
+            button.setImage(UIImage(named: asset), for: .normal)
+        }
+    }
+
+    private var isEnabled: Bool = true {
+        didSet {
+            button.isEnabled = isEnabled
+        }
+    }
+
+    private lazy var titleLabel: UILabel = {
+        let label = UILabel(withAutoLayout: true)
+        label.font = UIFont.captionStrong.withSize(12).scaledFont(forTextStyle: .footnote)
+        label.adjustsFontForContentSizeCategory = true
+        label.textAlignment = .center
+        label.text = heading
+        label.textColor = .textSecondary
+        return label
+    }()
+
+    private lazy var button: UIButton = {
+        let button = UIButton(withAutoLayout: true)
+        button.titleLabel?.font = UIFont.bodyStrong.withSize(17).scaledFont(forTextStyle: .footnote)
+        button.titleLabel?.adjustsFontForContentSizeCategory = true
+
+        let spacing = .smallSpacing / 2
+
+        button.semanticContentAttribute = .forceRightToLeft
+        button.imageEdgeInsets = UIEdgeInsets(top: spacing, leading: spacing, bottom: 0, trailing: -spacing)
+        button.titleEdgeInsets = UIEdgeInsets(top: 0, leading: -spacing, bottom: 0, trailing: spacing)
+        button.addTarget(self, action: #selector(handleButtonTap), for: .touchUpInside)
+
+        let buttonColor: UIColor = .btnPrimary
+        let interfaceBackgroundColor: UIColor = .bgPrimary
+        button.setTitleColor(buttonColor, for: .normal)
+        button.setTitleColor(buttonColor.withAlphaComponent(0.5), for: .highlighted)
+        button.setTitleColor(buttonColor.withAlphaComponent(0.5), for: .selected)
+        button.setTitleColor(.btnDisabled, for: .disabled)
+        button.tintColor = buttonColor
+
+        if heading != nil {
+            button.contentEdgeInsets = UIEdgeInsets(
+                top: titleLabel.font.pointSize,
+                leading: .mediumLargeSpacing + spacing,
+                bottom: 0,
+                trailing: .mediumLargeSpacing + spacing
+            )
+        }
+
+        return button
+    }()
+
+    private var heading: String?
+
+    // MARK: - Init
+
+    public init(heading: String) {
+        self.heading = heading
+        super.init(frame: .zero)
+        setup()
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setup()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    // MARK: - Setup
+
+    public var title: String? {
+        didSet {
+            button.setTitle(title, for: .normal)
+        }
+    }
+
+    private func setup() {
+        translatesAutoresizingMaskIntoConstraints = false
+        arrowDirection = .down
+        isEnabled = true
```

Comment:
These two are already given the default values at the top of the file

Reply:
Yes but if I don't set them it won't trigger the "didSet" call, and if I remove the default value then it will complain.

Any suggestions?

Reply:
The button will be enabled by default, so we can skip that one, but one solution for the arrow:

```swift
public var arrowDirection: ArrowDirection = .down {
    didSet { updateArrowDirection() }
}

// […]

private func setup() {
    updateArrowDirection()
    // Other code
}

// […]

private func updateArrowDirection() {
    let asset: FinniversImageAsset = arrowDirection == .up ? .arrowUpSmall : .arrowDownSmall
    button.setImage(UIImage(named: asset), for: .normal)
}
```

Reply:
I guess that's kind of the same but different. 

Works for me tho!